### ISSUE TITLE
Mono: -V switch does not display version on homebrew installation

### ISSUE THREAD

**Issue by eatdrinksleepcode** [scriptcs/scriptcs#1055]
Running "scriptcs -V" on Mono displays the version ASCII art but does not actually output the version number.

Environment:
OS X 10.10.2
Mono 4.0.1
ScriptCS 0.14.1

Mac/Mono:

```
[~]$ scriptcs -V                                                                                                                                                                          [ruby-2.1.1]
               _       _
 ___  ___ _ __(_)_ __ | |__ ___ ___
/ __|/ __| '__| | '_ \| __// __/ __|
\__ \ (__| |  | | |_) | |_| (__\__ \
|___/\___|_|  |_| .__/ \__\\___|___/
                |_| Version:
```

Windows/.NET:

```
$ scriptcs -V
               _       _
 ___  ___ _ __(_)_ __ | |__ ___ ___
/ __|/ __| '__| | '_ \| __// __/ __|
\__ \ (__| |  | | |_) | |_| (__\__ \
|___/\___|_|  |_| .__/ \__\\___|___/
                |_| Version: 0.14.1
```

**Comment by adamralph** [comment]
@eatdrinksleepcode thanks for raising! I guess it should be an easy fix...

**Comment by BlackFrog1** [comment]
In version 0.15, it displays 0.0.0
<img width="579" alt="Version information via Mono on OS X" src="https://cloud.githubusercontent.com/assets/550820/9146445/d25575b4-3d2a-11e5-83c6-bb8564f9c3c4.png">

**Comment by mbaker-dev** [comment]
I'll take it!

**Comment by glennblock** [comment]
Great!

**Comment by adamralph** [comment]
#1080 implies the problem is due to the build not working correctly in Mono, which I assume manifests itself when installing on Mac with homebrew? Is that the case?

What about the binaries which have been built on Windows (e.g. the ones published to Chocolatey)? Do the display the same problem when running on Mac/Linux?

**Comment by glennblock** [comment]
-v works for me with .svm

**Comment by adamralph** [comment]
Must be specific to homebrew then.

We should add an acceptance test for showing the version before closing this issue. That would catch this on Travis.

**Comment by mbaker-dev** [comment]
I haven't looked into the homebrew installation process, but the issue occurs when using a version of scriptcs that was built from build.sh.

The version that is displayed from -v is set from the `UpdateProjectVersions` build task in Build.proj. This was not being executed from build.sh, which used ScriptCs.sln instead of Build.proj. I updated  `RestoreCommonVersionInfo` in my pull request to be compatible with xbuild and updated build.sh to call it and `UpdateProjectVersions` so the version is set correctly.

**Comment by mbaker-dev** [comment]
@adamralph Could you provide guidance on how an acceptance test could be written for this? The build version is dynamically generated based on some complex conditions in a csproj file. Otherwise, my pull request fixes this issue on builds performed via build.sh.

**Comment by filipw** [comment]
Fixed with #1194
This works correctly off the dev branch at the moment.
